File: DeJia/app/src/main/java/com/dejia/anju/api/UpLoadImgApi.java
package com.dejia.anju.api;

import android.content.Context;

import com.dejia.anju.api.base.BaseCallBackListener;
import com.dejia.anju.api.base.BaseCallBackUploadApi;
import com.dejia.anju.net.NetWork;
import com.dejia.anju.net.ServerCallback;
import com.dejia.anju.net.ServerData;
import com.lzy.okgo.model.HttpParams;

import java.util.Map;

//上传头像接口
public class UpLoadImgApi implements BaseCallBackUploadApi {
    @Override
    public void getCallBack(Context context, Map<String, Object> maps, HttpParams uploadPamas, final BaseCallBackListener listener) {
        NetWork.getInstance().call("user", "uploadImg", maps, uploadPamas, new ServerCallback() {
            @Override
            public void onServerCallback(ServerData mData) {
                listener.onSuccess(mData);
            }
        });
    }
}

File: DeJia/app/src/main/java/com/dejia/anju/api/UpLoadImgApi.kt
package com.dejia.anju.api

import android.content.Context
import com.dejia.anju.api.base.BaseCallBackListener
import com.dejia.anju.api.base.BaseCallBackUploadApi
import com.dejia.anju.net.NetWork.Companion.instance
import com.dejia.anju.net.ServerCallback
import com.dejia.anju.net.ServerData
import com.lzy.okgo.model.HttpParams

//上传头像接口
class UpLoadImgApi : BaseCallBackUploadApi {
    override fun getCallBack(context: Context, maps: MutableMap<String, Any>, uploadPamas: HttpParams, listener: BaseCallBackListener<*>) {
        instance!!.call("user", "uploadImg", maps, uploadPamas, context, ServerCallback { mData: ServerData? -> listener.onSuccess(mData) })
    }
}